### PR TITLE
Refactor progress bar

### LINKED ISSUE #172: Add options to disable/hide progress bar
**Is your feature request related to a problem? Please describe.**

This is a minor thing, but I find the progress bar annoying when I run inference with pipeline successively.

See this screenshot for example.

In this case, I generated 10 images using DDIMPipeline and used `tqdm` myself, but the progress bars coming from `__call__` of the pipeline are stacking up and annoying.

**Describe the solution you'd like**

It would be nice if `disable` and `leave` options of `tqdm` is available with pipelines. Keyword arguments (say, `disable_tqdm` and `leave_tqdm`?) could be added to `__call__` methods and passed to `tqdm`. These are the relevant lines in case of `DDPMPipeline`:
https://github.com/huggingface/diffusers/blob/92b6dbba1a25ed27b0bae38b089715132c7e6bcc/src/diffusers/pipelines/ddpm/pipeline_ddpm.py#L31
https://github.com/huggingface/diffusers/blob/92b6dbba1a25ed27b0bae38b089715132c7e6bcc/src/diffusers/pipelines/ddpm/pipeline_ddpm.py#L47

I found [something that might be relevant](https://github.com/huggingface/diffusers/blob/92b6dbba1a25ed27b0bae38b089715132c7e6bcc/src/diffusers/utils/logging.py) in the logging module, but it's not used in the pipeline modules. Maybe this can be used.

### PR DESCRIPTION
**Pull request by hysts**
Close #172 

Hi, @patrickvonplaten 
I think https://github.com/huggingface/diffusers/commit/fda63082d5ae2e3906ca0fd2f57c4c55441b0086 is what you initially had in mind. But I think it would be better if we could set other `tqdm` parameters as well. It's because I think it would be better to use `leave=False` instead of disabling `tqdm` entirely sometimes, especially when using the DDPM scheduler with 1000 steps to generate images.
So I suggest https://github.com/huggingface/diffusers/commit/3aa44e8e86d9d27cc4ff70ff4294a21e5b1f326c. What do you think?

Usage:
```python
pipeline = DDPMPipeline.from_pretrained('CompVis/ldm-celebahq-256')
pipeline.to('cuda:0')

# if one wants to set `leave=False`
pipeline.set_progress_bar_config(leave=False)

# if one wants to disable `tqdm`
pipeline.set_progress_bar_config(disable=True)
```

BTW, I'm not familiar with the design of the `diffusers` yet and have a question. I added `__init__` to `DiffusionPipeline` class, but is this OK, or is there any other better/supposed way?

### CODE REVIEW

**Comment by hysts** [comment]
Oh, I just noticed https://github.com/huggingface/diffusers/issues/172#issuecomment-1225598380. But I guess this is already addressed in my PR?

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py [hunk 1]
```diff
@@ -126,7 +125,7 @@ def __call__(
         if accepts_eta:
             extra_step_kwargs["eta"] = eta
 
-        for i, t in tqdm(enumerate(self.scheduler.timesteps)):
+        for i, t in enumerate(self.progress_bar(self.scheduler.timesteps)):
```
Comment (hysts) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py:128]: As for `enumerate` and `tqdm`, if the order is `enumerate(tqdm(...))`, we don't have to pass `total`.

**Comment by hysts** [comment]
I think https://github.com/huggingface/diffusers/pull/226 and https://github.com/huggingface/diffusers/pull/236 are related to this PR and they would conflict with this.

File: src/diffusers/pipeline_utils.py [hunk 2]
```diff
@@ -57,6 +58,10 @@ class DiffusionPipeline(ConfigMixin):
 
     config_name = "model_index.json"
 
+    def __init__(self):
```
Comment (patrickvonplaten) [inline src/diffusers/pipeline_utils.py:61]: Could we remove the `__init__` function here? It's usually much easier to read components when there is no `__init__` function :-)

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py (hunk 1 above)
Comment (patrickvonplaten) [reply src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py:128]: Agree that's what it states in the `tqdm` docs: https://pypi.org/project/tqdm/#faq-and-known-issues

File: src/diffusers/pipeline_utils.py (hunk 2 above)
Comment (patrickvonplaten) [reply src/diffusers/pipeline_utils.py:61]: The main reason here is that adding an `__init__` to a class makes the class automatically harder to understand e.g. if you abstract from this class and the class doesn't have an init you're given much more freedom with your "parent" class

File: tests/test_pipelines.py [hunk 3]
```diff
@@ -44,6 +44,29 @@
...
+        up_block_types=("AttnUpBlock2D", "UpBlock2D"),
+    )
+    scheduler = DDPMScheduler(num_train_timesteps=10)
+
+    ddpm = DDPMPipeline(model, scheduler).to(torch_device)
+    ddpm(output_type="numpy")["sample"]
+    captured = capsys.readouterr()
+    assert "10/10" in captured.err, "Progress bar has to be displayed"
+
+    ddpm.set_progress_bar_config(disable=True)
```
Comment (patrickvonplaten) [inline tests/test_pipelines.py:64]: simply way to turn off the progress bar

**Review by patil-suraj — APPROVED** [review]
LGTM, thanks a lot for the PR @hysts ❤️

File: tests/test_pipelines.py [hunk 4]
```diff
@@ -44,6 +44,29 @@
...
+        in_channels=3,
+        out_channels=3,
+        down_block_types=("DownBlock2D", "AttnDownBlock2D"),
+        up_block_types=("AttnUpBlock2D", "UpBlock2D"),
+    )
+    scheduler = DDPMScheduler(num_train_timesteps=10)
+
+    ddpm = DDPMPipeline(model, scheduler).to(torch_device)
+    ddpm(output_type="numpy")["sample"]
+    captured = capsys.readouterr()
```
Comment (anton-l) [inline tests/test_pipelines.py:61]: Nice test, gonna remember this ;)

**Review by anton-l — APPROVED** [review]
LGTM, thanks for working on this @hysts @patrickvonplaten

**Comment by patrickvonplaten** [comment]
Sorry for having fiddled into your PR quite a bit here @hysts - thanks a mille for the contribution :heart:

**Pull request merged**